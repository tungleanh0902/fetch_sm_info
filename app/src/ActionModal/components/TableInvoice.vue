<template>
  <div>
    <ul class="table-invoice">
      <li v-if="subTotal > 0" class="sub-total">
        <span>Subtotal</span>
        <span> {{ subTotal | fullDecimals }} {{ transactionDenom }} </span>
      </li>
      <li class="fees">
        <span>Network Fee</span>
        <span>
          {{ fee.amount | fullDecimals }}
          {{ fee.denom !== transactionDenom ? fee.denom : transactionDenom }}
        </span>
      </li>
      <li class="total-row">
        <span>Total</span>
        <div class="total-column">
          <p>{{ total | fullDecimals }} {{ transactionDenom }}</p>
          <p v-if="fee.denom !== transactionDenom">
            {{ fee.amount | fullDecimals }} {{ fee.denom }}
          </p>
        </div>
      </li>
    </ul>
  </div>
</template>
<script>
import { fullDecimals } from "../../scripts/num"

export default {
  name: `table-invoice`,
  filters: {
    fullDecimals,
  },
  props: {
    amount: {
      type: Number,
      required: true,
    },
    fee: {
      type: Object,
      required: true,
    },
    transactionDenom: {
      type: String,
      required: true,
    },
  },
  data: () => ({
    info: `Estimated network fees based on simulation.`,
  }),
  computed: {
    subTotal() {
      return this.amount
    },
    total() {
      // if there is a feeDenom, it means that subTotal and estimatedFee are different currencies and
      // cannot be therefore added up together
      return this.fee.denom !== this.transactionDenom
        ? this.subTotal
        : Number(this.fee.amount) + this.subTotal
    },
  },
}
</script>
<style scoped>
.table-invoice {
  margin: 2rem 0 0;
  border-collapse: inherit;
  padding: 0 0.25rem;
  font-size: var(--sm);
  letter-spacing: 0.4px;
}

.table-invoice li {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.table-invoice span {
  padding: 0;
  color: var(--dim);
}

.table-invoice span:not(:first-child) {
  text-align: right;
}

.total-row {
  border-top: 2px solid var(--bc);
  margin-top: 0.5rem;
  padding-top: 0.25rem;
}

.total-column {
  display: flex;
  flex-direction: column;
}
</style>
