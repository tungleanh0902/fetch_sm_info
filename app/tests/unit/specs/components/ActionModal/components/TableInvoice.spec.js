import { shallowMount, createLocalVue } from "@vue/test-utils"
import TableInvoice from "src/ActionModal/components/TableInvoice"

describe(`TableInvoice`, () => {
  let wrapper
  const localVue = createLocalVue()
  localVue.directive(`tooltip`, () => {})

  beforeEach(() => {
    wrapper = shallowMount(TableInvoice, {
      localVue,
      propsData: {
        amount: 17.2,
        fee: {
          amount: 0.030864,
          denom: `STAKE`,
        },
        transactionDenom: `STAKE`,
      },
    })
  })

  it(`should render correctly`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should display the correct subtotal`, async () => {
    expect(wrapper.text()).toMatch(/17.2[0]+ STAKE/)
  })

  it(`should display the correct network fee`, async () => {
    expect(wrapper.text()).toMatch(/0.03086[0-9]+/)
  })

  it(`should display the correct total`, async () => {
    expect(wrapper.text()).toMatch(/17.23086[0-9]+ STAKE/)
  })
})
