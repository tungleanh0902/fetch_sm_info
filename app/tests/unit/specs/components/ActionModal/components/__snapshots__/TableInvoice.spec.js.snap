// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableInvoice should render correctly 1`] = `
<div>
  <ul
    class="table-invoice"
  >
    <li
      class="sub-total"
    >
      <span>
        Subtotal
      </span>
       
      <span>
         17.200000 STAKE 
      </span>
    </li>
     
    <li
      class="fees"
    >
      <span>
        Network Fee
      </span>
       
      <span>
        
        0.030864
        STAKE
      
      </span>
    </li>
     
    <li
      class="total-row"
    >
      <span>
        Total
      </span>
       
      <div
        class="total-column"
      >
        <p>
          17.230864 STAKE
        </p>
         
        <!---->
      </div>
    </li>
  </ul>
</div>
`;
